src/GodotOnReady.Generator/GodotOnReadySourceGenerator.cs

```diff
@@ -48,6 +48,10 @@ public void Execute(GeneratorExecutionContext context)
 
 			var resourceSymbol = GetSymbolByName("Godot.Resource");
 			var nodeSymbol = GetSymbolByName("Godot.Node");
+			var nodePathSymbol = GetSymbolByName("Godot.NodePath");
+
+			var dictionarySymbol = GetSymbolByName("System.Collections.Generic.Dictionary`2");
+			var nodeCacheSymbol = dictionarySymbol.Construct(nodePathSymbol, nodeSymbol);
```

Comment:
Now that I see this... I don't think `NodePath` actually works as a key. It isn't equatable, and Godot gives us a new instance for each property even if the strings are the same, so the cache always ends up missing. I tried this out because I wasn't sure:

```cs
[Export] public NodePath same1, same2; // Both point to the same LineEdit node in the editor
...
GD.Print(same1);                          // LineEdit
GD.Print(same2);                          // LineEdit
GD.Print(same1 == same2);                 // False
GD.Print((string)same1 == (string)same2); // True (cast implementation calls into engine code)
```

Could use `Dictionary<string, Node>` instead, to use the `(string)` result as the key. But, I'm actually not sure what this will do to performance. The `string` cast calls into the engine, and the engine has to actually concatenate some strings to determine the value! I'm not sure how the overhead of the `string` cast + `Dictionary` lookup compares to simply calling `GetNode` every time.

Since I'm not really sure if this lookup ends up helping, I think I would prefer to keep the implementation simple for now, and just call `GetNodeOrNull` every time. Sorry for the flip flop on this. 😅 

Let me know if you have some more thoughts on this--I haven't done any real perf work with Godot, I've just heard that the Mono -> native boundary is fairly expensive in itself.

Reply:
Yes I'm totally with you there, using this cache here would also lead to a performance loss for all the cases when there's only a single node. I guess this could be worked around by introducing yet another check and only generating the cache when there's actually more than 2 nodes, but that would further increase the complexity and even then the performance gained (if any at all) would probably be negligible.

I also though about creating a static cache by generating a variable based on the NodePath, but that would fail when the NodePath get's changed using the Editor, plus it would be an even more complicated solution.

Ultimately I think just getting the node multiple times is the best solution, especially since I'd imagine you only have 3 or 4 references to the same node at most anyway.